Add warnings, as requested by ISE

wkumari opened the pull request:
Eliot had requested, in his ISE review, that the document contains recommendations and guidance on when **not** to use a disconnected namespace. This provides that....

wkumari committed:
Add warnings, as requested by ISE

smutt commented:
Unless 'closed network' is a common term I would prefer we stick with 'private network'. I'm unfamiliar with the term 'closed network'.

Line 85 seems like it should be deleted. Something is wonky there.

On line 93 I disagree with the addition of the word 'currently'. If something changes WRT DNSSEC and .internal we should update this RFC. The word 'currently' only confuses the reader.

Line 142 has two shoulds.

wkumari commented:
> Unless 'closed network' is a common term I would prefer we stick with 'private network'. I'm unfamiliar with the term 'closed network'.
> 

"closed network" is from the original / I didn't change this, so it isn't really "stick with". 
Context: "The "internal" top-level domain provides this purpose in the DNS. Such domains will not resolve in the global DNS, but can be configured within closed networks as the network operator sees fit."

I'm fine with either - Kim?

> Line 85 seems like it should be deleted. Something is wonky there.

Indeed. No idea what happened there. Fixed.

> 
> On line 93 I disagree with the addition of the word 'currently'. If something changes WRT DNSSEC and .internal we should update this RFC. The word 'currently' only confuses the reader.

I think we need to further discuss this - the plan (from discussions with the ISE, and DNSOP chairs, and others) is that this will likely go back to DNSOP, and, along with Joe Abley's Delegation to Nowhere, will be that this will eventually work.

> 
> Line 142 has two shoulds.

Context: "users should should be aware that collisions"..
Well, users **really** should be aware :-)
Thanks, fixed...

wkumari committed:
Address Andrew's comments - https://github.com/kjd/draft-davies-internal-tld/pull/15#issuecomment-3611076713

wkumari committed:
Stupid typo (appears -> appear)

smutt commented:
RE "Closed network" it is a term that I just didn't know about.

https://www.google.com/search?hl=en&q=site%3Adatatracker.ietf.org%2Fdoc%20%22closed%20network%22

I don't think the definition from RFC 8595 matches our usage, but the other usages align well with our intention. I would still prefer we use 'private network' only so that we're consistent. But I'm not overly fussed about it.

We do need to discuss the word 'currently' on line 93 more. I see this as unrelated to the question of whether or not the status quo remains in place (i.e., .internal remains non-existent in the DNS root) or it changes (i.e., an unsigned insecure .internal appears in the DNS root). This is simply an editorial issue for me. If this is going to be an RFC the word 'currently' means nothing to someone who will be reading it 10 years from now. Recognizing that the Internet changes over time, there is always an implicit 'currently' in everything that the IETF publishes. This is why RFCs need updates sometimes.

wkumari committed:
Addressed more of Andrew's comment...

wkumari commented:
> RE "Closed network" it is a term that I just didn't know about.
> 
> https://www.google.com/search?hl=en&q=site%3Adatatracker.ietf.org%2Fdoc%20%22closed%20network%22
> 
> I don't think the definition from RFC 8595 matches our usage, but the other usages align well with our intention. I would still prefer we use 'private network' only so that we're consistent. But I'm not overly fussed about it.

I deleted the first occurrence ("such as within closed corporate or home networks") and made the second one be "private". I sort of think that the first one was helpful to differentiate things like ".internal." from things like "something.dc.icann.org", but "closed" still probably isn't the right word for that... 

> 
> We do need to discuss the word 'currently' on line 93 more. I see this as unrelated to the question of whether or not the status quo remains in place (i.e., .internal remains non-existent in the DNS root) or it changes (i.e., an unsigned insecure .internal appears in the DNS root). This is simply an editorial issue for me. 
[SNIP]
Okey dokey, fair enough --- I removed it.

I thought that you were meaning the "meta" / status quo question. 

Thank you!

kjd commented on draft-davies-internal-tld.mkd:
```diff
@@ -60,34 +60,35 @@ private applications.
-closed networks as the network operator sees fit. It fulfils a similar
-purpose as private-use IP address ranges that are set aside (e.g.
-{{RFC1918}}).
+private networks. It fulfills a purpose similar to private-use IP address
```
Is there is a reason to remove "as the network operator sees fit"? I think it is useful to specify who has agency over these decisions.

wkumari replied:
Done

kjd commented on draft-davies-internal-tld.mkd:
```diff
@@ -60,34 +60,35 @@ private applications.
-purpose as private-use IP address ranges that are set aside (e.g.
-{{RFC1918}}).
+private networks. It fulfills a purpose similar to private-use IP address
+ranges (e.g. {{RFC1918}}), but in the DNS space.
```
"space" seems superfluous here.

wkumari replied:
Done

kjd commented on draft-davies-internal-tld.mkd:
```diff
@@ -60,34 +60,35 @@ private applications.
 # Using the "internal" Namespace
 
-Network operators have been using different names for private-use
+Network operators have been using various,unregistered names for private-use
```
s/,/ /

kjd replied:
or "... using a variety of unregistered names..." may read better

wkumari replied:
Thank you -- I took your second suggestion,

kjd commented on draft-davies-internal-tld.mkd:
```diff
@@ -60,34 +60,35 @@ private applications.
+of this decision, including:
+1. The potential for collisions if multiple networks using "internal" are
+   interconnected in the future
+2. The risk of leakage of "internal" names into the global DNS, and t
```
dangling "t" at the end of this line?

kjd commented on draft-davies-internal-tld.mkd:
```diff
@@ -60,34 +60,35 @@ private applications.
 
-DNSSEC validating resolvers relying on the global DNS trust anchor will
-fail to resolve names ending in "internal".
+DNSSEC validating resolvers relying on the global DNS trust anchor will fail to resolve names ending in "internal".
```
Should this be item 4 in the list above?

kjd commented on draft-davies-internal-tld.mkd:
```diff
@@ -136,6 +137,14 @@ be many different servers named "accounting.internal". Users should be
 aware of this when performing operations requiring authenticity, such as
 entering credentials.
 
+Related to the lack of uniqueness, users should be aware that collisions
```
s/Related to/Given/

kjd reviewed (commented):
I've added some comments in-line, generally fine with the spirit of the changes.

wkumari committed:
Attemped to resolve Kim's comments.

Note: I'm trying Github Codespace for the first time, and I have no idea if it is working correctly....

wkumari commented:
Thank you; I believe that I have addressed all of the comments. Can you please check and submit a new version if OK?

wkumari commented:
Thank you — would you also post a new version of the draft please?

It expires in a few days —
https://datatracker.ietf.org/doc/draft-davies-internal-tld/

W


On Wed, Dec 31, 2025 at 2:36 PM, Kim Davies ***@***.***>
wrote:

> Merged #15 <https://github.com/kjd/draft-davies-internal-tld/pull/15>
> into main.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kjd/draft-davies-internal-tld/pull/15#event-21812546898>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDG3XHMDELVXBGJPIPDIO34EQQTVAVCNFSM6AAAAACNTKKAGWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMRRHAYTENJUGY4DSOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

wkumari commented:
(I'm not a contributor to the GitHub repo so I can't easily do it myself)


On Wed, Dec 31, 2025 at 3:02 PM, Warren Kumari ***@***.***> wrote:

> Thank you — would you also post a new version of the draft please?
>
> It expires in a few days —
> https://datatracker.ietf.org/doc/draft-davies-internal-tld/
>
> W
>
>
> On Wed, Dec 31, 2025 at 2:36 PM, Kim Davies ***@***.***>
> wrote:
>
>> Merged #15 <https://github.com/kjd/draft-davies-internal-tld/pull/15>
>> into main.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kjd/draft-davies-internal-tld/pull/15#event-21812546898>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABDG3XHMDELVXBGJPIPDIO34EQQTVAVCNFSM6AAAAACNTKKAGWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMRRHAYTENJUGY4DSOA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***
>> com>
>>
>
>
